### ISSUE TITLE
aws_route_table bug with ignore_changes - "diffs didn't match during apply"

### ISSUE THREAD

**Issue by carlsverre** [hashicorp/terraform#3869]
All output summarized here: https://gist.github.com/carlsverre/f50395eaee55952d95ed

Here is the repro (also in the gist):

```
provider "aws" {
    region = "us-east-1"
}

resource "aws_vpc" "default" {
    cidr_block = "172.17.0.0/24"
}

resource "aws_subnet" "default" {
    vpc_id = "${aws_vpc.default.id}"
    cidr_block = "172.17.0.0/24"
}

resource "aws_internet_gateway" "default" {
    vpc_id = "${aws_vpc.default.id}"
}

resource "aws_route_table" "default" {
    vpc_id = "${aws_vpc.default.id}"

    route {
        cidr_block = "0.0.0.0/0"
        gateway_id = "${aws_internet_gateway.default.id}"
    }
    lifecycle {
        ignore_changes = ["route"]
    }
}
```

If you comment the ignore_changes line it applies fine.

**Comment by carlsverre** [comment]
* Note: I originally had create_before_destroy specified as well, this was unneeded and doesn't affect the repo so I removed it for clarity.

**Comment by dverbeek84** [comment]
I have the same issue. Only on initial run and destroy.

**Comment by hanninen** [comment]
Same here, `diff's didn't match during apply` error is shown on each `terraform apply`, when `ignore_changes` is on for the `aws_route_table` resource.

**Comment by mgoodness** [comment]
This directive is a requirement for the way I'm provisioning a Kubernetes cluster with AWS cloud provider support. Kubernetes adjusts the route table at startup to suit its needs. Without ```ignore_changes```, TF resets those changes.

**Comment by phinze** [comment]
Hi folks, some causes for `diffs didn't apply` on `ignore_changes` were fixed up in #4965. Others will be tackled in #5627 by me. Head to #5627 if you're still having issues on v0.6.12.

**Comment by carlsverre** [comment]
Awesome - thanks phinze!  Since original report I ended up going a different direction that didn't require ignore_changes.  Will loop back if I encounter any more issues with it.
